File: app/src/main/java/ru/vtb/ppcd/example/processor/impl/NoopMessageProcessor.java
package ru.vtb.ppcd.example.processor.impl;

import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.AllArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import ru.vtb.ppcd.example.processor.ExchangeContext;
import ru.vtb.ppcd.example.processor.MessageProcessorFunction;
import ru.vtb.ppcd.generated.async.model.CreateProductRequestInnerEvent;

import java.util.UUID;


@Slf4j
@Service
@AllArgsConstructor
public class NoopMessageProcessor
        implements MessageProcessorFunction<CreateProductRequestInnerEvent, CreateProductRequestInnerEvent> {
    private final ObjectMapper objectMapper;

    @Override
    public CreateProductRequestInnerEvent processMessage(
            ExchangeContext<CreateProductRequestInnerEvent> exchangeContext
    ) {
        log.info("NoopMessageProcessor.exchangeContext: {} {}", exchangeContext, UUID.randomUUID());
        return exchangeContext.getMessage();
    }

    @Override
    public Class<CreateProductRequestInnerEvent> processClass() {
        return CreateProductRequestInnerEvent.class;
    }

    @Override
    public ObjectMapper getObjectMapper() {
        return objectMapper;
    }
}

File: ms-ppcd-example-app/src/main/kotlin/ru/vtb/ppcd/example/processor/impl/NoopMessageProcessor.kt
package org.example.ru.vtb.ppcd.example.processor.impl

import com.fasterxml.jackson.databind.ObjectMapper
import io.github.oshai.kotlinlogging.KotlinLogging
import java.util.*
import org.example.ru.vtb.ppcd.example.model.ExchangeContext
import org.example.ru.vtb.ppcd.example.processor.MessageProcessorFunction
import org.springframework.stereotype.Service
import ru.vtb.ppcd.generated.async.model.CreateProductRequestInnerEvent

private val logger = KotlinLogging.logger {}

@Service
class NoopMessageProcessor(
    private val objectMapper: ObjectMapper
) : MessageProcessorFunction<CreateProductRequestInnerEvent, CreateProductRequestInnerEvent> {
    override fun processMessage(
        exchangeContext:
        ExchangeContext<CreateProductRequestInnerEvent>
    ): CreateProductRequestInnerEvent {
        logger.info { "NoopMessageProcessor.exchangeContext - $exchangeContext, operationUUID - ${UUID.randomUUID()}" }
        return exchangeContext.message
    }

    override fun processClass() = CreateProductRequestInnerEvent::class.java

    override fun getObjectMapper() = objectMapper
}